File: Baekjoon/Class3/28.18870 좌표 압축.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const xn = input[1].split(" ").map(Number);
const sortedXn = Array.from(new Set(xn)).sort((a, b) => a - b); // 중복 제거, 정렬

const compression = new Map(); // Map으로 순서 저장
sortedXn.forEach((sortedX, i) => compression.set(sortedX, i));

const out = xn.map(x => compression.get(x));
console.log(out.join(" "));

File: Baekjoon/Class3/28.18870 좌표 압축.py
#dict 사용
import sys
input = sys.stdin.readline

input()
xn = list(map(int, input().split()))
#중복 제거, 정렬, dict로 순서 저장
compression = {sorted_x: i for i, sorted_x in enumerate(sorted(set(xn)))}
print(*(compression[x] for x in xn))

''' 이진 탐색
from bisect import bisect_left
import sys
input = sys.stdin.readline

input()
xn = list(map(int, input().split()))
sorted_xn = sorted(set(xn)) #중복 제거, 정렬
print(*(bisect_left(sorted_xn, x) for x in xn))
'''